javalib/src/main/scala/java/util/stream/Collector.scala

```diff
@@ -0,0 +1,126 @@
+package java.util.stream
+
+import java.util.{Collections, HashSet, Set}
+import java.util.function._
+
+trait Collector[T, A, R] {
+
+  def accumulator(): BiConsumer[A, T]
+
+  def characteristics(): Set[Collector.Characteristics]
+
+  def combiner(): BinaryOperator[A]
+
+  def finisher(): Function[A, R]
+
+  def supplier(): Supplier[A]
+}
+
+object Collector {
+
+  /* Design Note:
+   *   enums are a tangle in each of Java and Scala.
+   *   To make things more fun, enum changes between Scala 2 and Scala 3.
+   *
+   *   Scala Native currently supports both Scala 2 & 3. There are mechanisms
+   *   for handling code which must vary by Scala version.
+   *
+   *   This is an acknowledgedly incomplete & inaccurate Scala 2 style
+   *   emulation of the parts of the Java Collector.Characteristics
+   *   enum needed to implement the Collectors class. It is intended to
+   *   hit the Pareto point of benefit/cost.
+   *
+   *   Of course, somebody, somewhere, is going to use one of the features
+   *   not emulated.  Then it will be time for a better iteration.
+   */
+
+  case class Characteristics(name: String)
+
+  object Characteristics {
+
+    final val CONCURRENT = new Characteristics("CONCURRENT")
+    final val IDENTITY_FINISH = new Characteristics("IDENTITIY_FINISH")
+    final val UNORDERED = new Characteristics("UNORDERED")
+
+    def valueOf(name: String): Collector.Characteristics = {
+      name match {
+        case nom if (nom == "CONCURRENT") => Characteristics.CONCURRENT
+        case nom if (nom == "UNORDERED")  => Characteristics.UNORDERED
+        case nom if (nom == "IDENTITY_FINISH") =>
+          Characteristics.IDENTITY_FINISH
+        case _ =>
+          val missingName =
+            s"java.util.stream.Collector.Characteristics.${name}"
+          throw new IllegalArgumentException(s"No enum constant ${missingName}")
+      }
+    }
+
+    def values(): Array[Collector.Characteristics] = {
+      // Fill the array in the same order as shown by the JVM.
+      // Since an Array is mutable, create & fill on each call, rather than
+      // returning a static array.
+      val va = new Array[Collector.Characteristics](3)
+      va(0) = Characteristics.CONCURRENT
+      va(1) = Characteristics.UNORDERED
+      va(2) = Characteristics.IDENTITY_FINISH
+
+      va
+    }
+  }
```

Comment:
Recently I've found a way to satisfy both Scala 2 and Scala 3 enums, by using `extends _Enum[T]`. Thanks to the name override (which would be fixed in compiler plugin, similarly to the _Object, _Class), Scala 3 compiler would not give errors. Here's example: https://github.com/scala-native/scala-native/blob/ac411cf5c52ca8febd06d1dad2b2630b911e6fb2/javalib/src/main/scala/java/lang/Thread.scala#L358-L379

This would also make sure that type check `_.isInstanceOf[Enum[_]]` would pass in runtime.

Reply:
This is the major ugliness.  Thank you for the suggestion.  I will take a run at implementing it.